### PR TITLE
!welcome by Hess

### PR DESCRIPTION
**Pull request by BHess2653**
!welcome command.

sends welcome message to new users and a message to the general chat.
use !welcome to resend the welcome message.

### CODE REVIEW

**Comment by brennondenny** [comment]
Everything works fine for me, maybe consider changing the welcome command to !tos or change it's description in the help menu to say it is sending the tos since that's the main reason a user would use that command

File: bot/client.js [hunk 1]
```diff
@@ -57,9 +57,16 @@ client.on('message', (message) => {
   }
 });
 
-// If User joins server this message will be sent
+// Event listener for new members
 client.on('guildMemberAdd', (member) => {
-  member.sendMessage(`${member.user.username}, Welcome to the Full Sail Armada! These are the terms of service...`);
+  // Send welcome message to members who join
+  member.sendMessage(`${member}, Welcome to the Full Sail Armada! These are the terms of service...`);
```
Comment (mattpezzente) [inline bot/client.js]: Small detail, but consider lower-casing the "W" in "Welcome" as it comes after a comma

Comment (mattpezzente) [inline bot/client.js]: "sendMessage" is being deprecated, please change those methods to just "send" before continuing.

**Review by mattpezzente — CHANGES_REQUESTED** [review]
See comments on commit, some deprecation changes should be made.

File: bot/controllers/welcome.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+const BaseController = require('../baseController.js');
+const Command = require('../baseCommand.js');
+
```
Comment (brennondenny) [inline bot/controllers/welcome.js]: Add in the default comments found across the controllers to your welcome.js file.

**Review by brennondenny — CHANGES_REQUESTED** [review]
Add in the default comments found across the controllers to your welcome.js file.

**Review by brennondenny — APPROVED** [review]

**Review by mattpezzente — APPROVED** [review]

File: bot/client.js [hunk 3]
```diff
@@ -57,4 +57,16 @@ client.on('message', (message) => {
...
 });
 
+// Event listener for new members
+client.on('guildMemberAdd', (member) => {
+  // Send welcome message to members who join
+  member.send(`${member}, Welcome to the Full Sail Armada! These are the terms of service...`);
+  // Send message to specific channel on the server
+  const channel = member.guild.channels.find('name', 'general');
+  // If channel is not found do nothing
+  if (!channel) return;
```
Comment (reactivepixel) [inline bot/client.js]: This is a non-intuitive way to 'do nothing'.

File: bot/client.js [hunk 4]
```diff
@@ -57,4 +57,16 @@ client.on('message', (message) => {
   }
 });
 
+// Event listener for new members
+client.on('guildMemberAdd', (member) => {
+  // Send welcome message to members who join
+  member.send(`${member}, Welcome to the Full Sail Armada! These are the terms of service...`);
```
Comment (reactivepixel) [inline bot/client.js]: Duplicate logic in your controller.

File: bot/client.js [hunk 5]
```diff
@@ -57,4 +57,16 @@ client.on('message', (message) => {
...
+  // Send message to specific channel on the server
+  const channel = member.guild.channels.find('name', 'general');
+  // If channel is not found do nothing
+  if (!channel) return;
+  // Send message with members name
+  channel.send(`Welcome to the Full Sail Armada ${member}! for more information use !help`);
```
Comment (reactivepixel) [inline bot/client.js]: The announcement to the general channel should not have a !help cmd reminder. This should be in the welcome message if anywhere.

File: bot/controllers/welcome.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    // Aliasing 'this' as controller to allow for binding in actions
+    const controller = this;
+
+    // Array of all commands, see baseCommand.js for prototype
+    this.commands = [
+      new Command(
+        '!tos',
+        '!tos',
+        'Terms of Service',
+        'Resends terms of service message.',
```
Comment (reactivepixel) [inline bot/controllers/welcome.js:19]: Good phrasing on your help message

**Review by reactivepixel — CHANGES_REQUESTED** [review]
Very good work incorporating the feedback from your peers in shaping this submission. Ideally, I would have like you to solve the duplicate code logic issue out of the gate, but all submissions seem to have missed this, and we will discuss this further on Wednesday. The idea there would be to invoke the controller you have built rather than manually sending a message in the guildadduser event. Either way, solid work!

**Pull request closed without merging**